Composer/packages/tools/language-servers/language-generation/src/LGServer.ts

```diff
@@ -553,6 +561,38 @@ export class LGServer {
       return Promise.resolve(null);
     }
 
+    const wordRange = getEntityRangeAtPosition(document, params.position);
+    const word = document.getText(wordRange);
+    const projectId = this.getLGDocument(document)?.projectId;
+
+    const getEntities = async () => {
+      let suggestEntities: string[] = [];
+      if (projectId && this.entitiesResolver) {
+        const luContents = this.entitiesResolver(projectId);
+        if (luContents) {
+          for (const content of luContents) {
+            const luisJson = await extractLUISContent(content);
+            suggestEntities = suggestEntities.concat(getSuggestionEntities(luisJson, suggestionAllEntityTypes));
+          }
+        }
+      }
+
+      this.luisEntities = suggestEntities;
+    };
+
+    setTimeout(getEntities);
```

Comment:
I think if this will take a long time, we can move it to the lgWorker.ts

Reply:
moved to lgworker to do the lu file parse